File: ps-frontend/projects/ps-lib/src/model/contract-specification.ts
import { ContractSpecificationItem } from "./contract-specification-item";

export class ContractSpecification {
  public id: number;
  public description: string;
  public dateCreated?: Date;
  public dateStart?: Date;
  public dateEnd?: Date;
  public active?: boolean;

  public contractSpecificationItems?: ContractSpecificationItem[];
}

File: ps-backend/src/main/java/org/commonground/ps/backendapi/model/ContractSpecification.java
package org.commonground.ps.backendapi.model;

import java.util.Date;
import java.util.List;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class ContractSpecification {
  private Long id;
  private String description;
  private Date dateCreated;
  private Date dateStart;
  private Date dateEnd;
  private Boolean active;

  private List<ContractSpecificationItem> contractSpecificationItems;
}
